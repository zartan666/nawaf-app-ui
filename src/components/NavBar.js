import React from 'react';
import {StyleSheet, View} from 'react-native';

const NavBar = ({style, left, right, middle}) => (
  <View style={[styles.navBar, style]}>
    <View style={styles.left}>
      {left}
    </View>
    <View style={styles.middle}>
      {middle}
    </View>
    <View style={styles.right}>
      {right}
    </View>
  </View>
);

NavBar.propTypes = {
  left: React.PropTypes.object,
  middle: React.PropTypes.object,
  right: React.PropTypes.object,
};

const styles = StyleSheet.create({
  navBar: {
    top: 0,
    left: 0,
    right: 0,
    height: 64,
    backgroundColor: '#EFEFF2',
    paddingTop: 20,
    flexDirection: 'row',
  },
  left: {},
  middle: {
    flex: 1,
  },
  right: {},
});

export default NavBar;
