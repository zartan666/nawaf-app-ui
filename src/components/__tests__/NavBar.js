import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './../NavBar';
import {Text} from 'react-native';

test('renders null', () => {
  const tree = renderer
    .create(
      <NavBar
        left={<Text>left icon</Text>}
        middle={<Text>title</Text>}
        right={<Text>right icon</Text>}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
