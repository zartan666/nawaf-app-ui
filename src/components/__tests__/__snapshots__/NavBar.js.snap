// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders null 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EFEFF2",
        "flexDirection": "row",
        "height": 64,
        "left": 0,
        "paddingTop": 20,
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      left icon
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      title
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      right icon
    </Text>
  </View>
</View>
`;
